xds/src/test/java/io/grpc/xds/FallbackLbTest.java

```diff
@@ -0,0 +1,377 @@
+    List<EquivalentAddressGroup> eags22 = ImmutableList.of(eag221);
+    String lbConfigRaw22 = "{"
+        + "\"balancerName\" : \"dns:///balancer.example.com:8080\","
+        + "\"fallbackPolicy\" : [{\"fallback_2\" : { \"fallback_2_option\" : \"no\"}}]"
+        + "}";
+    @SuppressWarnings("unchecked")
+    Map<String, ?> lbConfig22 = (Map<String, ?>) JsonParser.parse(lbConfigRaw22);
+    fallbackLb.handleResolvedAddresses(ResolvedAddresses.newBuilder()
+        .setAddresses(eags22)
+        .setAttributes(Attributes.newBuilder().set(ATTR_LOAD_BALANCING_CONFIG, lbConfig22).build())
+        .build());
+
+    verify(balancer2).handleResolvedAddresses(ResolvedAddresses.newBuilder()
+        .setAddresses(eags22)
+        .setAttributes(Attributes.newBuilder()
+            .set(ATTR_LOAD_BALANCING_CONFIG, ImmutableMap.of("fallback_2_option", "no")).build())
+        .build());
+
+    assertThat(helpers1).hasSize(1);
+    assertThat(balancers1).hasSize(1);
+    assertThat(helpers2).hasSize(1);
+    assertThat(balancers2).hasSize(1);
+
+    verify(balancer2, never()).shutdown();
+    fallbackLb.shutdown();
+    verify(balancer2).shutdown();
+  }
+
+
+  @Test
+  public void fallback_handleBackendsEagsOnly() throws Exception {
+    EquivalentAddressGroup eag0 = new EquivalentAddressGroup(
+        ImmutableList.<SocketAddress>of(new InetSocketAddress(8080)));
+    Attributes attributes = Attributes
+        .newBuilder()
+        .set(GrpcAttributes.ATTR_LB_ADDR_AUTHORITY, "this is a balancer address")
+        .build();
+    EquivalentAddressGroup eag1 = new EquivalentAddressGroup(
+        ImmutableList.<SocketAddress>of(new InetSocketAddress(8081)), attributes);
+    EquivalentAddressGroup eag2 = new EquivalentAddressGroup(
+        ImmutableList.<SocketAddress>of(new InetSocketAddress(8082)));
+    List<EquivalentAddressGroup> eags = ImmutableList.of(eag0, eag1, eag2);
+
+    String lbConfigRaw = "{"
+        + "\"balancerName\" : \"dns:///balancer.example.com:8080\","
+        + "\"fallbackPolicy\" : [{\"fallback_1\" : { \"fallback_1_option\" : \"yes\"}}]"
+        + "}";
+    @SuppressWarnings("unchecked")
+    Map<String, ?> lbConfig = (Map<String, ?>) JsonParser.parse(lbConfigRaw);
+    fallbackLb.handleResolvedAddresses(ResolvedAddresses.newBuilder()
+        .setAddresses(eags)
+        .setAttributes(Attributes.newBuilder().set(ATTR_LOAD_BALANCING_CONFIG, lbConfig).build())
+        .build());
+
+    LoadBalancer balancer1 = balancers1.get(0);
+    verify(balancer1).handleResolvedAddresses(
+        ResolvedAddresses.newBuilder()
+            .setAddresses(ImmutableList.of(eag0, eag2))
+            .setAttributes(
+                Attributes.newBuilder()
+                    .set(ATTR_LOAD_BALANCING_CONFIG, ImmutableMap.of("fallback_1_option", "yes"))
+                    .build())
+            .build());
+  }
+
+  @Test
+  public void fallback_onlyGrpclbAddresses_NoBackendAddress() throws Exception {
```

Comment:
nit: maybe name it "incompatibleWithPureGrpclbAddresses"?

Reply:
It's not clear what `incompatible` mean. I think the existing name is more meaningful.